runtime/src/main/kotlin/kotlin/collections/Arrays.kt

```diff
@@ -972,6 +1060,54 @@ public actual fun LongArray.contentHashCode(): Int {
     return result
 }
 
+/**
+ * Returns a hash code based on the contents of this array as if it is [List].
+ */
+@SinceKotlin("1.3")
+@ExperimentalUnsignedTypes
+public fun UByteArray.contentHashCode(): Int {
+    var result = 1
+    for (element in this)
+        result = 31 * result  + element.hashCode()
```

Comment:
Why call `element.hashCode()` and not just `element`?

Reply:
1) Kotlin doesn't have mixed signed-unsigned operations.
2) All other variants of this function call hashCode, including one for ints.